await-wrap.js

```diff
@@ -0,0 +1,151 @@
+const thenify = require('tinythen');
+const toString = Object.prototype.toString;
+
+/**
+ * The beginning experiment for async/await support for Datastar models.
+ * We use this class to wrap each model and the used methods with thenables.
+ * We also create a new method so we can still support streaming for findAll
+ *
+ * @class AwaitWrap
+ */
+class AwaitWrap {
+  /**
+   * Initialize instance with model
+   *
+   * @param {Dynastar} model A defined dynamodb model wrapper Dynastar
+   * @constructor
+   */
+  constructor(model) {
+    this.model = model;
+    this._hoist(model);
+  }
+  /**
+   * Hoist any labeled hoistable functions found on the Dynastar compatibility wrapper
+   *
+   * @param {Dynastar} model A defined dynamodb model wrapper Dynastar
+   */
+  _hoist(model) {
+    if (model.hoistable && model.hoistable.length) {
+      for (const name of model.hoistable) {
+        const fn = model[name];
+        // Here we basically assume that unless its already an async function
+        //  its callback based and we thenify it
+        if (toString.call(fn).toLowerCase().slice(8, -1) !== 'asyncfunction') {
+          this[name] = function () {
+            return thenify(model, name, ...arguments);
```

Comment:
any reason for `thenify` over the (now) built in [`util.promisify`](https://nodejs.org/dist/latest-v12.x/docs/api/util.html#util_util_promisify_original) (added in `node@8`) ?

Reply:
@msluther it just doesnt have the same timing things as a promise. @bmeck knows more. Im not partial either way.